src/streamsync/ui_manager.py

```diff
@@ -0,0 +1,70 @@
+from typing import Union
+
+from streamsync.core import StreamsyncSession, session_manager
+from streamsync.core_ui import (Component, SessionComponentTree, UIError,
+                                current_parent_container)
+
+
+class StreamsyncUI:
+    """Provides mechanisms to manage and manipulate UI components within a
+    Streamsync session.
+
+    This class offers context managers and methods to dynamically create, find,
+    and organize UI components based on a structured component tree.
+    """
+
+    def __init__(self, session_id: str):
+        self.session: Union[StreamsyncSession, None] = session_manager.get_session(session_id)
+        if self.session is None:
+            raise RuntimeError("Invalid session passed to the UI manager")
+        # Initialize the component tree with the session
+        self.component_tree: SessionComponentTree = \
+            self.session.session_component_tree
+        self.root_component = self.component_tree.get_component("root")
+        if not self.root_component:
+            raise RuntimeError(f"Failed to acquire root component in session {session_id}")
+
+    @staticmethod
+    def assert_in_container():
+        container = current_parent_container.get(None)
+        if container is None:
+            raise UIError("A component can only be created inside a container")
+
+    @property
+    def root(self) -> Component:
+        if not self.session:
+            raise RuntimeError("No session object on UI manager")
+        if not self.root_component:
+            raise RuntimeError(f"Failed to acquire root component in session {self.session.session_id}")
+        return self.root_component
+
+    def find(self, component_id: str) \
```

Comment:
It can work as one due to the return being a component, which supports context management. I did it so that users could also directly assign components to variables and modify properties, in line with what @ramedina86 was suggesting. If there's some extra benefits from adding this decorator that I'm overlooking, please let me know.

Reply:
Sorry I didn't push the complete implementation on this one:

```python
@contextlib.contextmanager
def find(self, component_id: str) -> Generator[Component, None, None]:
    # Example context manager for finding components
    component = self.component_tree.get_component(component_id)
    if component is None:
        raise RuntimeError(f'Component {component_id} not found')
    yield component
```

Reply:
Ok, I didn't know it was working without it.